src/app/services/global-api-http.service.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import {
+  SpectatorService,
+  createServiceFactory,
+} from '@ngneat/spectator/jest';
+import { WINDOW } from 'app/helpers/window.helper';
+import { GlobalApiHttpService } from './global-api-http.service';
+
+describe('GlobalApiHttpService', () => {
+  let spectator: SpectatorService<GlobalApiHttpService>;
+  let httpMock: HttpTestingController;
+
+  const mockWindow = {
+    location: {
+      protocol: 'https:',
+    },
+  } as unknown as Window;
+
+  const createService = createServiceFactory({
+    service: GlobalApiHttpService,
+    imports: [HttpClientTestingModule],
+    providers: [
+      { provide: WINDOW, useValue: mockWindow },
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createService();
+    httpMock = spectator.inject(HttpTestingController);
+  });
+
+  it('should call getBootId() and return string', () => {
+    const expectedBootId = 'boot-abc';
+
+    spectator.service.getBootId().subscribe((result) => {
+      expect(result).toBe(expectedBootId);
```

Comment:
You can put anything in `toBe` and test will pass. You need to wait for returning value, probably with something like `await firstValueFrom(...)`